firebase-firestore/src/main/java/com/google/firebase/firestore/local/StatsCollector.java

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+/**
+ * Collects the operation count from the persistence layer. Implementing classes can expose this
+ * information to measure the efficiency of persistence operations.
+ *
+ * <p>The only consumer of the operation counts is currently the LocalStoreTestCase (via {@link
+ * com.google.firebase.firestore.local.AccumulatingStatsCollector}). If you are not interested in
+ * the stats, you can use `newNoOpStatsCollector()` to return an empty stats collector.
+ */
+abstract class StatsCollector {
+  /** A stats collector that discards all operation counts. */
+  static class NoOpStatsCollector extends StatsCollector {
```

Comment:
This could be `private static` if `newNoOpStatsCollector()` had the return type of `StatsCollector`.

Alternatively, just consider making `StatsCollector` non-abstract and `new StatsCollector()` would be enough to get you a no-op instance. That would cut the line count here and reduce the number of classes in production by one.

Reply:
Are your really gonna make me change this PR from a "sophisticated stats framework" to a dumb counter? Okay then. It's done.